File: src/main/java/net/sjrx/intellij/plugins/systemdunitfiles/filetypes/AbstractUnitFileType.java
package net.sjrx.intellij.plugins.systemdunitfiles.filetypes;

import com.intellij.lang.Language;
import com.intellij.openapi.fileTypes.LanguageFileType;
import org.jetbrains.annotations.Nls;
import org.jetbrains.annotations.NotNull;

public abstract class AbstractUnitFileType extends LanguageFileType {

  protected AbstractUnitFileType(@NotNull Language language) {
    super(language);
  }
  
  protected AbstractUnitFileType(@NotNull Language language, boolean secondary) {
    super(language, secondary);
  }
  
  @Nls
  @Override
  public abstract @NotNull String getDisplayName();
  
}

File: src/main/kotlin/net/sjrx/intellij/plugins/systemdunitfiles/filetypes/AbstractUnitFileType.kt
package net.sjrx.intellij.plugins.systemdunitfiles.filetypes

import com.intellij.lang.Language
import com.intellij.openapi.fileTypes.LanguageFileType
import org.jetbrains.annotations.Nls

abstract class AbstractUnitFileType : LanguageFileType {
  protected constructor(language: Language) : super(language)
  protected constructor(language: Language, secondary: Boolean) : super(language, secondary)

  abstract override fun getDisplayName(): @Nls String
}
